src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,194 @@
+    private static final String USER_PASSWORD_OK = "123456";
+    private static final String USER_PASSWORD_THREE_CHARS = "123";
+    private static final String USER_PASSWORD_FIVE_CHARS = "12345";
+    private static final String USER_PASSWORD_LENGTH_OK = "1234567";
+    private static final int MIN_PASSWORD_LENGTH = 6;
+    private static RegistrationService registrationService;
+    private static User user;
+    private static User newUser;
+
+    @BeforeAll
+    static void beforeAll() {
+        registrationService = new RegistrationServiceImpl();
+    }
+
+    @BeforeEach
+    void setUp() {
+        user = new User();
+        user.setLogin(USER_NAME);
+        user.setAge(AGE_18);
+        user.setPassword(USER_PASSWORD_OK);
+
+        newUser = new User();
+        newUser.setLogin(USER_NAME);
+        newUser.setAge(AGE_18);
+        newUser.setPassword(USER_PASSWORD_OK);
+    }
+
+    @Test
+    void register_nullUser_notOk() {
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(null);
+        });
+    }
+
+    @Test
+    void register_emptyLogin_notOk() {
+        user.setLogin(null);
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    void register_nullLogin_notOk() {
+        user.setLogin(null);
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    void register_sameLogin_notOk() {
+        newUser.setLogin(USER_NAME);
+        assertEquals(newUser.getLogin(), user.getLogin());
+    }
+
+    @Test
+    void register_rightLogin_ok() {
+        user.setLogin(NEW_USER_NAME);
+        assertNotEquals(newUser.getLogin(), user.getLogin());
+    }
+
+    @Test
+    void register_nullAge_notOk() {
+        user.setAge(null);
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    void register_negativeAge_notOk() {
+        user.setAge(NEGATIVE_AGE);
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    void register_underAge_notOk() {
+        user.setAge(AGE_0);
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(user);
+        });
+        user.setAge(AGE_5);
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(user);
+        });
+        user.setAge(AGE_10);
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(user);
+        });
+        user.setAge(AGE_15);
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    void register_overAge_notOk() {

```

Comment:
delete **all test methods** where you do not call  `registrationService.register();`
I can't understand what are they written for?